### PR TITLE
feat: align namespaced shares deserialization with latest celestia 

### LINKED ISSUE #21: Update Celestia to the latest main and report the bugs found

### PR DESCRIPTION
**Pull request by zvolin**
closes #21

### CODE REVIEW

File: rpc/tests/share.rs [hunk 1]
```diff
@@ -63,10 +63,20 @@ async fn test_get_shares_by_namespace_wrong_ns(client: &HttpClient) {
...
-    client
-        .share_get_shares_by_namespace(&dah, random_ns())
+    let random_ns = random_ns();
+    let ns_shares = client
+        .share_get_shares_by_namespace(&dah, random_ns)
         .await
-        .unwrap_err();
+        .unwrap();
+
+    if !ns_shares.rows.is_empty() {
```
Comment (oblique) [inline rpc/tests/share.rs:72]: is this valid? if yes add a comment with brief explanation.

File: types/src/share.rs [hunk 2]
```diff
@@ -57,6 +57,32 @@ impl Share {
...
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(transparent)]
+struct RawNamespacedShares {
+    rows: Option<Vec<NamespacedRow>>,
+}
+
+impl From<RawNamespacedShares> for NamespacedShares {
+    fn from(value: RawNamespacedShares) -> Self {
+        Self {
+            rows: value.rows.unwrap_or(vec![]),
```
Comment (oblique) [inline types/src/share.rs:69]:
```suggestion
            rows: value.rows.unwrap_or_default(),
```

File: types/src/share.rs [hunk 3]
```diff
@@ -57,6 +57,32 @@ impl Share {
...
+    }
+}
+
+impl From<NamespacedShares> for RawNamespacedShares {
+    fn from(value: NamespacedShares) -> Self {
+        let rows = if !value.rows.is_empty() {
+            Some(value.rows)
+        } else {
+            None
+        };
```
Comment (oblique) [inline types/src/share.rs:80]: in this case we can avoid `not`

```suggestion
        let rows = if value.rows.is_empty() {
            None
        } else {
            Some(value.rows)
        };
```

File: types/src/share.rs [hunk 4]
```diff
@@ -151,23 +187,107 @@ mod tests {
...
+        assert_eq!(
+            max_ns_bytes,
+            b64_decode("//////////////////////////////////////8=")
+        );
+        assert_eq!(
+            hash_bytes,
+            b64_decode("OBQQFb/BaYJ+fBd9qWCox8r2wzLXrzLddHN3BjWOllg=")
+        );
+
+        let nmt_rs::NamespaceProof::AbsenceProof { leaf: Some(leaf), .. } = proof.deref() else {
```
Comment (oblique) [inline types/src/share.rs:242]:
```suggestion
        let nmt_rs::NamespaceProof::AbsenceProof { leaf: Some(leaf), .. } = &*proof else {
```

File: rpc/tests/share.rs (hunk 1 above)
Comment (zvolin) [reply rpc/tests/share.rs:79]: it is, explained

File: rpc/tests/share.rs [hunk 5]
```diff
@@ -63,10 +64,34 @@ async fn test_get_shares_by_namespace_wrong_ns(client: &HttpClient) {
         .unwrap()
         .dah;
 
-    client
-        .share_get_shares_by_namespace(&dah, random_ns())
+    let random_ns = random_ns();
```
[the remark this answers is not in the document]
Comment (zvolin) [reply rpc/tests/share.rs:67]: do you want this to be the two separate tests or just have both those cases ensured just in single test?

[the remark this answers is not in the document]
Comment (zvolin) [reply rpc/tests/share.rs:67]: asking as blob submitting is slow

**Review by oblique — APPROVED** [review]

**Pull request merged**